### PR TITLE
adding accuracy, attack and defense bonuses and 1-handed penalty

### PR DESCRIPTION
**Pull request by fabio-t**

### CODE REVIEW

**Comment by fabio-t** [comment]
I'm not sure if you are still playing URW.. but your tool is still pretty much the best. There's another one, but the source code has been lost - which is bad for future versions.

So I updated yours a bit.

File: Program.cs [hunk 1]
```diff
@@ -49,10 +49,16 @@ class UrwObject {
 		public float valuePerLbs;
 		public byte quality;
 
+		public byte h1Penalty;
+
+		public byte attackBonus;
+		public byte defenceBonus;
```
Comment (smilediver) [inline Program.cs]: Nitpicking but it's "defense" :)

File: Program.cs [hunk 2]
```diff
@@ -100,10 +116,18 @@ class UrwObject {
...
 			this.quality = (byte)(data.ReadByte() & 0x0F);
 
-			data.ReadBytes(6); // Unknown
+			data.ReadBytes(1); // Unknown
+
+			if (type == UrwObjectType.Weapon) {
+				int combatBonuses = data.ReadByte();
+				this.attackBonus = (byte)(combatBonuses & 0x0F);
+				this.defenceBonus = (byte)(combatBonuses >> 4);
+			}
```
Comment (smilediver) [inline Program.cs]: Forgot "else data.ReadByte();"? Cause I'd assume you need to skip a byte for non weapons.

**Review by smilediver — CHANGES_REQUESTED** [review]

**Comment by smilediver** [comment]
I play it on and off. :) If you can fix that else thing, I'd be glad to accept the changes. Awesome stuff otherwise!

File: Program.cs (hunk 1 above)
Comment (fabio-t) [reply Program.cs]: Depends where you live! I'm more of a British English guy, he :P

File: Program.cs (hunk 2 above)
Comment (fabio-t) [reply Program.cs]: You are right, I will add it. That "if" was a last minute addition (superfluous really) which I hadn't tested.

**Comment by fabio-t** [comment]
This time I tested it, should be OK ;)

**Pull request merged**